HW03/Denis_Havanski/spec/den_havanski_hw_03_t_03_spec.rb

```diff
@@ -0,0 +1,71 @@
+
+RSpec.describe 'task3' do
+  describe '#return_action_duration' do
+
+    let(:log_normal) do <<~INPUT
+      2018-04-23 17:17:49.7 ubuntu-xenial[14319] Debug - Calling core with action: event
+      2018-04-23 17:17:49.7 ubuntu-xenial[14319] Debug - connecting to: 10.6.246.101
+      2018-04-23 17:17:49.8 ubuntu-xenial[14319] Debug - docker event processed
+      2018-04-23 17:18:19.5 ubuntu-xenial[14319] Debug - monitoring grid communication health
+      2018-04-23 17:18:38.8 ubuntu-xenial[14319] Debug - Calling core with action: messages
+      2018-04-23 17:18:38.8 ubuntu-xenial[14319] Debug - connecting to: 10.6.246.101
+      2018-04-23 17:18:59.8 ubuntu-xenial[14319] Debug - inside docker_handle_event
+    INPUT
+    end
+
+    let(:log_more_two_val_events) do <<~INPUT
+      2018-04-23 17:17:49.7 ubuntu-xenial[14319] Debug - Calling core with action: event
+      2018-04-23 17:17:49.7 ubuntu-xenial[14319] Debug - connecting to: 10.6.246.101
+      2018-04-23 17:17:49.8 ubuntu-xenial[14319] Debug - Calling core docker event processed
+      2018-04-23 17:18:19.5 ubuntu-xenial[14319] Debug - monitoring grid communication health
+      2018-04-23 17:18:38.8 ubuntu-xenial[14319] Debug - Calling core with action: messages
+      2018-04-23 17:18:38.8 ubuntu-xenial[14319] Debug - connecting to: 10.6.246.101
+      2018-04-23 17:18:59.8 ubuntu-xenial[14319] Debug - Calling core inside docker_handle_event
+    INPUT
+    end
+
+    let(:log_only_one_val_event) do <<~INPUT
+      2018-04-23 17:17:49.7 ubuntu-xenial[14319] Debug - Calling core with action: event
+      2018-04-23 17:17:49.7 ubuntu-xenial[14319] Debug - connecting to: 10.6.246.101
+      2018-04-23 17:17:49.8 ubuntu-xenial[14319] Debug - docker event processed
+      2018-04-23 17:18:19.5 ubuntu-xenial[14319] Debug - monitoring grid communication health
+      2018-04-23 17:18:38.8 ubuntu-xenial[14319] Debug -  with action: messages
+      2018-04-23 17:18:38.8 ubuntu-xenial[14319] Debug - connecting to: 10.6.246.101
+      2018-04-23 17:18:59.8 ubuntu-xenial[14319] Debug - inside docker_handle_event
+    INPUT
+    end
+
+    let(:log_no_val_events) do <<~INPUT
+      2018-04-23 17:17:49.7 ubuntu-xenial[14319] Debug - core with action: event
+      2018-04-23 17:17:49.7 ubuntu-xenial[14319] Debug - connecting to: 10.6.246.101
+      2018-04-23 17:17:49.8 ubuntu-xenial[14319] Debug - Cal docker event processed
+      2018-04-23 17:18:19.5 ubuntu-xenial[14319] Debug - monitoring grid communication health
+      2018-04-23 17:18:38.8 ubuntu-xenial[14319] Debug -  with action: messages
+      2018-04-23 17:18:38.8 ubuntu-xenial[14319] Debug - connecting to: 10.6.246.101
+      2018-04-23 17:18:59.8 ubuntu-xenial[14319] Debug - inside docker_handle_event
+    INPUT
+    end
+
+    context 'when there are no valid events or only one' do
+      it 'return "0"' do
+        expect( task3(log_no_val_events) == "0" && task3(log_only_one_val_event) == "0").to be_truthy
```

Comment:
```suggestion
        expect(task3(log_no_val_events)).to eq(0)
```
you can split this check into two contexts and two checks

Reply:
I agree, corrected it. At one stage of writing tests, I thought it was a good idea to combine checks into boolean expressions. But then I realized that it wasn’t: this causes the problem of understanding which scenario the mistake was made in. I fixed it everywhere, I did not notice it here.